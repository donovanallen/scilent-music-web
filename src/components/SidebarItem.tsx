'use client';

import { Tooltip } from '@nextui-org/react';
import { useMediaQuery } from '@uidotdev/usehooks';
import Link from 'next/link';
import React from 'react';
import { IconType } from 'react-icons';

import { cn } from '@/lib/utils';

type SidebarItemProps = {
  icon?: IconType;
  label: string;
  active?: boolean;
  href: string;
  disabled?: boolean;
};

const SidebarItem: React.FC<SidebarItemProps> = ({
  icon: Icon,
  label,
  active,
  href,
  disabled = false,
}) => {
  const isMediumDevice = useMediaQuery('only screen and (min-width : 769px)');

  return (
    <Tooltip
      content={label}
      key={label}
      shadow='md'
      size='lg'
      classNames={{
        content: 'text-brand-primary bg-dark h4',
      }}
      placement='right'
      isDisabled={isMediumDevice}
    >
      <Link
        href={href}
        className={cn(
          `flex flex-row h-auto items-center w-full md:gap-x-4 cursor-pointer hover:text-brand-primary transition text-light py-1`,
          active && 'text-brand-dark',
          disabled &&
            'text-neutral-700 cursor-not-allowed hover:text-neutral-600',
        )}
        aria-disabled={disabled}
      >
        {Icon && <Icon size={26} />}
        <h4 className='hidden md:flex truncate w-100'>{label}</h4>
      </Link>
    </Tooltip>
  );
};

export default SidebarItem;
