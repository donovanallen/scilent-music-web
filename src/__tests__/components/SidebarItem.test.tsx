import { render } from '@testing-library/react';
import React from 'react';

import SidebarItem from '@/components/SidebarItem';

describe('SidebarItem', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<SidebarItem label='Test' href='' />);
    expect(asFragment()).toMatchSnapshot();
  });
});
