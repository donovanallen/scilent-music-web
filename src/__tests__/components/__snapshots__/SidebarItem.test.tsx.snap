// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem renders correctly 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    class="flex flex-row h-auto items-center w-full gap-x-4 cursor-pointer hover:text-brand-primary transition text-light py-1"
    href=""
  >
    <h4
      class="truncate w-100"
    >
      Test
    </h4>
  </a>
</DocumentFragment>
`;
